DigitalLearningSolutions.Data.Tests/Services/CustomPromptsServiceTests.cs

```diff
@@ -1,174 +1,288 @@
+                (new List<(int, string)> { (1, promptName) });
+
+            // When
+            var result = customPromptsService.GetCustomPromptsAlphabeticalList();
+
+            // Then
+            A.CallTo(() => customPromptsDataService.GetCustomPromptsAlphabetical()).MustHaveHappened();
+            result.Contains((1, promptName)).Should().BeTrue();
+        }
+
+        [Test]
+        public void AddCustomPromptToCentre_add_prompt_at_expected_prompt_number()
+        {
+            // Given
+            A.CallTo
+            (
+                () => customPromptsDataService.AddCustomPromptToCentre(1, A<int>._, 1, true, null)
+            ).DoesNothing();
+            A.CallTo(() => customPromptsDataService.GetCentreCustomPromptsByCentreId(1))
+                .Returns
+                (
+                    CustomPromptsTestHelper.GetDefaultCentreCustomPromptsResult(
+                        customField1Prompt: "Prompt",
+                        customField2Prompt: "Prompt",
+                        customField3Prompt: null,
+                        customField4Prompt: null,
+                        customField5Prompt: null,
+                        customField6Prompt: null
+                    )
+                );
+
+            // When
+            var result = customPromptsService.AddCustomPromptToCentre(1, 1, true, null);
+
+            // Then
+            A.CallTo
+            (
+                () => customPromptsDataService.AddCustomPromptToCentre(1, 3, 1, true, null)
+            ).MustHaveHappened();
+            result.Should().BeTrue();
+        }
+
+        [Test]
+        public void AddCustomPromptToCentre_add_prompt_at_expected_prompt_number_with_gaps_in_prompt_numbers()
+        {
+            // Given
+            A.CallTo
+            (
+                () => customPromptsDataService.AddCustomPromptToCentre(1, A<int>._, 1, true, null)
+            ).DoesNothing();
+            A.CallTo(() => customPromptsDataService.GetCentreCustomPromptsByCentreId(1))
+                .Returns
+                (
+                    CustomPromptsTestHelper.GetDefaultCentreCustomPromptsResult(
+                        customField1Prompt: "Prompt",
+                        customField2Prompt: null,
+                        customField3Prompt: "Prompt",
+                        customField4Prompt: "Prompt",
+                        customField5Prompt: "Prompt",
+                        customField6Prompt: null
+                    )
+                );
+
+            // When
+            var result = customPromptsService.AddCustomPromptToCentre(1, 1, true, null);
+
+            // Then
+            A.CallTo
+            (
+                () => customPromptsDataService.AddCustomPromptToCentre(1, 2, 1, true, null)
+            ).MustHaveHappened();
+            result.Should().BeTrue();
+        }
+
+        [Test]
+        public void AddCustomPromptToCentre_add_prompt_with_all_prompts_defined()

```

Comment:
These tests are great, but I think we could improve the title namings a bit to make it clearer why the test fails/what the expected result is. For example:
AddCustomPromptToCentre_does_not_add_prompt_if_centre_has_all_prompts_defined
AddCustomPromptToCentre_add_prompt_at_lowest_possible_prompt_number_with_gaps_in_prompt_number

etc